src/core/thread/mle.cpp

```diff
@@ -1560,6 +1594,25 @@ void Mle::HandleStateChanged(otChangedFlags aFlags)
             (Get<KeyManager>().GetSecurityPolicyFlags() & OT_SECURITY_POLICY_NATIVE_COMMISSIONING) != 0);
     }
 
+#if OPENTHREAD_FTD && OPENTHREAD_CONFIG_BACKBONE_ROUTER_ENABLE
+    if ((aFlags & OT_CHANGED_THREAD_BACKBONE_ROUTER_STATE) &&
+        (Get<BackboneRouter::Local>().IsPrimary() !=
+         (mBackboneRouterPrimaryAloc.GetAddress().GetLocator() != Mac::kShortAddrInvalid)))
+    {
+        if (Get<BackboneRouter::Local>().IsPrimary())
+        {
+            mBackboneRouterPrimaryAloc.GetAddress().SetLocator(kAloc16BackboneRouterPrimary);
+            mBackboneRouterPrimaryAloc.GetAddress().SetPrefix(GetMeshLocalPrefix());
+            Get<ThreadNetif>().AddUnicastAddress(mBackboneRouterPrimaryAloc);
+        }
+        else
+        {
+            Get<ThreadNetif>().RemoveUnicastAddress(mBackboneRouterPrimaryAloc);
+            mBackboneRouterPrimaryAloc.GetAddress().SetLocator(Mac::kShortAddrInvalid);
```

Comment:
Why not keep the locator always set to `kAloc16BackboneRouterPrimary`? Addition/removal of address should be sufficient, right?

Reply:
Yes,  it should also be sufficient. 
Previous thought in doing this way is that each addition/remove would iterate through the unicast address list to check whether or not it is contained by comparing address pointer. Here uses the last two bytes content to directly check whether or not the address is already added, which I thought might be a bit more efficient? Thoughts?

Reply:
Yes, it should be sufficient and it might also be more simple.
Original thoughts is that as not per bbr state update would require addtion/removal for example from `disabled` to`secondary` or vice versa.
Here uses the last two bytes content to directly check whether or not the address is already added in advance to avoid unnecessary addition/removal as the unicast address list would always be iterated to check whether or not it is contained by comparing address pointer during addition/removal.

No strong opinion on this, I am fine to update to use the simpler way if you prefer it.

Reply:
I agree that execution cycles may be slightly reduced, but at the cost of increased code size. In general, I expect the addition/removal of addresses to occur relatively rarely. For that reason, I would have a preference for smaller code size. Thoughts?

Reply:
👍  make sense! will update.